### ISSUE TITLE
feat: API for user to set value for the NoItems render

### ISSUE THREAD

**Issue by jaymorelli96** [charmbracelet/bubbles#410]
**Is your feature request related to a problem? Please describe.**
At the moment I couldn't find a way to remove the default `"No " + m.itemNamePlural + " found."` that is set when there are no visible items in the list. 

**Describe the solution you'd like**
I would like an API in the model that I could do the following:
```go
l := list.New() // New list is created, I am omitting the params for simplicity 
l.SetNoItemsPlaceHolder("Currently this list is empty") // Accepts a string
l.SetNoItemsPlaceHolder() // Possibility to leave blank
```

**Describe alternatives you've considered**
I tried something like this, but I couldn't get the title bar to show as well. Also it seems to much of a effort that could be easily done with an API 😄 
```go
func (m Model) requestsView() string {
	items := m.requests.VisibleItems()

	if len(items) == 0 {
		noItems := m.requests.Styles.NoItems.Render("Nothing found.")
		return lipgloss.NewStyle().Height(m.height).Render(noItems)
	}

	return m.requests.View()
}
```

**Additional context**
This logic is already being done by the `func (m Model) populatedView() string` in the `list.go` line 1153. Should be simple to add a new `showNoItems bool` in the model for at least have the option to omit this

**Comment by carhartl** [comment]
Yes, please! It's also that with an empty list that also shows the status bar, you get duplicated rendering of the string "No items" (one in the status bar without a period, one in the empty list with a period). The fallback string in the list also seems to be misaligned (see screenshot).

(Initially was thinking that at least when filtering is enabled to not render a fallback at all, seemingly its redundant , but that won't work in all of the cases.)

<img width="418" alt="Screenshot 2024-10-23 at 16 03 58" src="https://github.com/user-attachments/assets/ade71b6f-5601-4048-8e76-d7105b014e54">

PS: Simple enough workaround, I'm stripping the string away in the models' `View()` method:

```go
s := strings.Replace(m.list.View(), "No repositories.", "", 1)
```
